src/main/java/seedu/address/storage/JsonAdaptedPerson.java

```diff
@@ -112,22 +112,20 @@ public Person toModelType() throws IllegalValueException {
         }
         final Name modelName = new Name(name);
 
-        if (company == null) {
-            throw new IllegalValueException(String.format(MISSING_FIELD_MESSAGE_FORMAT, Company.class.getSimpleName()));
-        }
-        if (!Company.isValidCompany(company)) {
-            throw new IllegalValueException(Company.MESSAGE_CONSTRAINTS);
+        if (!(company == null)) {
+            if (!Company.isValidCompany(company)) {
+                throw new IllegalValueException(Company.MESSAGE_CONSTRAINTS);
+            }
         }
```

Comment:
Can we avoid deep nesting here? We can combine the 2 `if` statements.
Same for job title as well.